pkg/chains/formats/slsa/internal/material/material.go

```diff
@@ -215,3 +229,55 @@ func RemoveDuplicateMaterials(mats []common.ProvenanceMaterial) ([]common.Proven
 	}
 	return out, nil
 }
+
+// AddMaterialsFromPipelineParamsAndResults extracts type hinted params and results and adds the url and digest to materials.
+func AddMaterialsFromPipelineParamsAndResults(ctx context.Context, pro *objects.PipelineRunObject, mats []common.ProvenanceMaterial) []common.ProvenanceMaterial {
+	sms := artifacts.RetrieveMaterialsFromStructuredResults(ctx, pro, artifacts.ArtifactsInputsResultName)
+	mats = append(mats, sms...)
+
+	var commit, url string
```

Comment:
I merged this into one function instead of splitting into two because everything is updating the commit and url string.

Reply:
SGTM